Ten simple ways to make it easier to re(use) your data
======================================================

Authors
-------


Themes
------
1. Make individual files as easy to understand and work with as possible
2. Make it as easy as possible to combine your data with other datasets
3. Make it easy to find and use your data

Introduction
------------

Sharing data is increasingly recognized as an important component of the
scientific process (). The sharing of scientific data is beneficial because it
allows the replication of the original research results, the reuse of the data
in meta-analyses and synthetic science, and the use of data in ways not
originally conceived of by the data collectors. In ecology and evolutionary
biology this sharing occurs through a combination of standardized repositories
like GenBank (http://www.ncbi.nlm.nih.gov/genbank/) and repositories that can
hold any of the diverse forms of ecological and evolutionary data like Dryad
(http://datadryad.org/), Ecological Archives
(http://esapubs.org/archive/default.htm), and Figshare (http://figshare.com).

While data sharing is increasingly common and straightforward, much of the
shared data is not easily reused because it does not follow best practices in
terms of data structure, metadata, and licensing [@jones2006]. This makes it
more difficult to work with existing data and therefore makes the data less
useful than it could be [@jones2006, @reichman2011].

Here we provide a list of 10 simple ways to make it easier to reuse the data
that you share. These recommendations focus on making your data available,
understandable, and easy to work with. This means that the recommendations not
only make it easier for others to use your data, but also make it easier
to work with your own data.


1. Share your data
------------------

Share your data.  If you only follow one tenet of our manuscript it should be this one.  We believe that data sharing benefits both the community and the data sharer.    Specifically, sharing data can increase citation numbers and recognition by peers in the scientific community [@Piwowar2007](http://www.plosone.org/article/info:doi%2F10.1371%2Fjournal.pone.0000308). These benefits will likely result in new collaborations and increased productivity. Shared datasets are also more easily reused in the future by the data sharer because they are documented and standardized. Additionally, although some researchers still resist data sharing, it is increasingly required for funding [@nsfpolicy](http://www.nsf.gov/bfa/dias/policy/dmp.jsp), [@nihpolicy](http://grants.nih.gov/grants/guide/notice-files/NOT-OD-03-032.html), [@nsercpolicy](http://www.nserc-crsng.gc.ca/Professors-Professeurs/FinancialAdminGuide-GuideAdminFinancier/Responsibilities-Responsabilites_eng.asp), [@fwfpolicy](http://www.fwf.ac.at/en/public_relations/oai/index.html), publications [@dryadmembers](https://www.datadryad.org/pages/members), [@Piwowar2008](http://ocs.library.utoronto.ca//index.php/Elpub/2008/paper/view/684), and potentially by law (e.g. FASTR [@FASTR](http://www.taxpayeraccess.org/action/FASTR_calltoaction.shtml)) [More info on FASTR](http://www.creativecommons.org/weblog/entry/36699).  Therefore, those that do not share will likely be left behind, and they will impede the rate of progress in their intellectual fields [@Piwowar2011](http://www.nature.com.dist.lib.usu.edu/nature/journal/v473/n7347/full/473285a.html), [@Parr2005](http://www.aseanbiodiversity.info/Abstract/51005017.pdf).

A survey of the Ecological Society of America suggests that reluctance to share data is largely due to perceived fears of: 1) competition for publications based on the shared data, 2) a lack of recognition for shared data, and 3) logistical barriers [@Palmer2004](http://www.esa.org/ecovisions/ppfiles/EcologicalVisionsReport.pdf). The first reason is largely short-sided because many data sharing initiatives allow for data embargos which can last for several years while the authors develop their publications and thus avoid competition for deriving publications from the data.  Additionally, data sets are citable entities and therefore recognition is provided to the data provider in the form of increased citation metrics. Lastly, logistical barriers to data sharing are diminishing as high profile archives for data and code are becoming more numerous and receiving greater attention [@Parr2005](http://www.aseanbiodiversity.info/Abstract/51005017.pdf).


2. Provide metadata
-------------------

The first key to using data is understanding what it is. Metadata is information about the data that enables long-term reuse of the data set by the original investigators and secondary use by other scientists. Datasets that might be relevant to another's work will likely be overlooked if the metadata is unclear or doesn't exist (Fraser & Gluck 1999, Zimmerman 2003). Metadata can take several forms, including descriptive names in the data files themselves, a written description of the data, and images (_i.e.,_ maps, photographs). Ideally, metadata should improve ease of use of the dataset through providing 1) the what, when, where, and how of data collection, 2) how to find and access the data, 3) suggestions on the suitability of the data for answering specific questions, and 4) warnings to users on known problems or inconsistencies in the data (Michener _et al._ 1997, Zimmerman 2003). 

Good metadata does not have to be associated with high monetary or time investment. Cultivating good habits of describing the data during planning and data collection stages can help keep the original investigator organized and make eventual publication of the data easier. Clear descriptive information on data collection methods and data structure (e.g., units of measurement, data type, linked columns) will help future users, including the original investigator, to understand if the data is appropriate to use for a new project and how to avoid biasing results by including uncertain or missing data. It is also critical that the metadata includes detailed description of problems or inconsistencies in the data that a future user should be aware of, such as missing data, mid-study changes to sampling regime, personnel turnover, habitat disturbance, change in environmental conditions, or data anomalies (Zimmerman 2003), as well as the specific identifier used to flag problematic data. 

Similar to any other scientific publication, metadata should be logically organized, complete and clear enough to enable interpretation and use of the data, and contain a statement on the overall quality and integrity of the data. Metadata that is poorly organized, incomplete, or buries important details in wordy or confusing text hinders secondary use or may lead to critical errors in future work (Zimmerman 2007). Specific metadata standards are being developed (_e.g.,_ EML, DIF, DC, FGDC, Ecological Archives metadata standard; Reichman _et al._ 2011, Whitlock 2011), but the most important thing is to _have_ metadata.

__citations__
* http://knb.ecoinformatics.org/eml_metadata_guide.html
* Fraser, B. and M. Gluck. 1999. Usability of geospatial metadata _or_ space-time matters. Bulletin of the American Society for Information Science. __25__:24-28.
* Michener, W. K. and J. W. Brunt. 2000. _Ecological Data: Design, Management and Processing_. Blackwell Science. Malden, MA.
* Michener, W. K., J. W. Brunt, J. J. Helly, T. B. Kirchner, and S. G. Stafford. 1997. Nongeospatial metadata for the ecological sciences. Ecological Applications. __7__:330-342
* Reichmann, O. J., M. B. Jones, and M. P. Schildhauer. 2011. Challenges and opportunities of open data in ecology. Science. __331__:703-705
* Whitlock, M. C. 2011. Data archiving in ecology and evolution: best practices. Trends in Ecology and Evolution. __26__:61-65.
* Zimmerman, A. S. 2003. Data sharing and secondary use of scientific data: Experiences of Ecologists. Dissertation. University of Michigan.
* Zimmerman, A. S. 2007. Not by metadata alone: the use of diverse forms of knowledge to locate data for reuse. International Journal on Digital Libraries. __7__:5-16.


3. Provide an unprocessed form of the data
------------------------------------------

Most often, the data that are presented in a manuscript table or even used in the analysis are modified in some way from the original form in which they were collected. Unprocessed data are data presented in the form that resulted directly from the data collection effort. It may be important here to distinguish between a methods protocol that processes samples to retrieve data and an analytical or computational processing that manipulates unprocessed data into useful summary statistics, standard indices or metrics for your analysis. As much as a model or mathematical correction might be best for your purposes, these sort of modifications to unprocessed data can make the difference for the reuse of your data.

This is not to say that your data are best suited for a meta-analysis in the raw form, but as unprocessed was the starting point for you so should that option be available to future work. Of course, your work to develop and process the data is equally important. Indeed, some metrics and indices are widely used across disciplines. In any case where data is modified from its original form or summarized (i.e. averaged) for an analysis or publication, we encourage that you also make this data available. Better yet, include computer scripts that process the data to the form you used alongside the unprocessed data. This way, any one can observe the process by which you arrived at the values used in your analysis.

One concern we anticipate is that unprocessed data files can be extremely large. Later on, we urge you to use a data repository, which often can store and share such large files. Otherwise, we recommend the .zip or .tar.gz compression formats to reduce the bit size of your data.
	

4. Use standard formats that will work across systems
------------------------------------------------------
Using standard formats makes your data directly compatible with different systems and software.  Both the format of the file itself and the contents of the file are important to take into consideration.  Most ecological data can be stored effectively in a .csv file.  This can be opened by any type of software, and this is a file type that never goes out of date.  Proprietary formats such as .xls and .xlsx are difficult to use for people not working with Microsoft Office, are often difficult to load into other programs, and can become obsolete, such that you may not be able to access your own data years later as the format becomes unsupported by newer versions of the software.

The second major consideration when applying standard formats is formatting the contents of the file.  Figure # illustrates some common formatting problems that can obstruct data use.  These errors can cause the data to become unusable without a great deal of reformatting, which obstructs use of the data both by the original collector of the data and others who might want to reuse the data later.

Figure # : standard_format_errors.pdf

5. Use standard data structures
-------------------------------

The structure of data can become a major barrier to reuse if it does not conform to widely recognized standards.  This is particularly true in ecology and evolution where the datasets can cover a wide variety of heterogeneous types of information.  Certain data types in ecology and evolution already have well established standard structures such as FASTA files for nucleotide or peptide sequences [@FASTA](http://zhanglab.ccmb.med.umich.edu/FASTA/) and the Newick phylogenetic tree format; however, this is generally not the case and here we will describe a set of general rules for structuring tabular data.  We focus on tabular data in this section simply because it is likely the most widely encountered data type in ecology and evolution, and it presents the data sharer with the most flexibility in structure and therefore has the potential to provide the data user with the most obstacles.
 
Tabular data refers to a regular two-dimensional array of rows and columns.  Within this structure, each row should represent a single unique data-record, and each column should refer a single type of information associated with each record.  {More to come here} 

* Note: Should we mention when a relational database is going to be needed rather than simple tabular data?
* Note: Should we comment on image or audio file formats?


6. Use good null values
-----------------------
Performing analyses on datasets with missing data can be problematic.  This can be compounded by the use of inappropriate null values.  Null values are characters that are placed into to a field to indicate the presence of missing data.  Unfortunately, there are many different ways to indicate a missing value, and very little agreement on which null value to use.

The null values that are most compatible with software commonly in use by ecologists are a blank, NULL, or NA.  Blanks are the most compatible across different software, and are easily spotted in a visual examination of the data.  However, there can be hidden spaces in a blank cell, so be certain that if you use this option, you check for hidden spaces during the quality control check or strip excess white space computationally.  If, for some reason, you are performing calculations with Microsoft Excel, be aware that Excel will treat blanks as if they were zeros.  If you are going to be working primarily in R, and are not going to be using a relational database, using NA makes the most sense, as long as you are not also using NA as an abbreviation for North America.  If you are working with SQL, a blank or NULL would be the best option.  Having to use a null value can sometimes be avoided by making sure that the data is structured in a standard database format.  Whichever null value that you choose to use, make sure that you only use one, and that you use it consistently throughout the data set.  In addition, indicate your choice of null value clearly in the metadata. 

<table>
<table border="1">
    <tr>
        <th>Null values	</td>
		<th>Problems</td>
		<th>Compatibility</td>
		<th>Recommendation</td>
	</tr>
	<tr>
		<td>0</td>
		<td>Indistinguishable from a true zero</td>
		<td></td>
		<td>Never use</td>
	</tr>
	<tr>
		<td></td>
		<td>Hard to distinguish if the value is not there because it was missing, or because it was collected, but overlooked on entry.  A  line that looks blank can contain a hidden space, which results in computational errors.  Counted as a zero for calculation purposes in Microsoft Excel.</td>
		<td>Will work with Python and SQL</td>
		<td>Good option, if not performing calculations in Microsoft Excel.</td>
	</tr>
	<tr>
		<td>999, -999, 9999, -9999</td>
		<td>Not recognized as a null value by many programs without user input, can be inadvertently entered into calculations.</td>
		<td></td>
		<td>Avoid</td>
	</tr>
	<tr>
		<td>NA, na</td>
		<td>Also an abbreviation for North America, can cause problems with data type (turn a numerical column into a text column).  NA is more commonly recognized than na.</td>
		<td>Used by R</td>
		<td>Good option</td>
	</tr>
	<tr>
		<td>N/A</td>
		<td>An alternate form of NA, but not machine readable.</td>
		<td></td>
		<td>Avoid</td>
	</tr>
	<tr>
		<td>NULL</td>
		<td>Can cause problems with data type (turn a numerical column into a text column)</td>
		<td>Used by SQL</td>
		<td>Good option</td>
	</tr>
	<tr>
		<td>None</td>
		<td>Can cause problems with data type (turn a numerical column into a text column)</td>
		<td>Recognized by Python</td>
		<td>Avoid</td>
	</tr>
	<tr>
		<td>No data</td>
		<td>Can cause problems with data type (turn a numerical column into a text column), contains a space</td>
		<td></td>
		<td>Avoid</td>
	</tr>
	<tr>
		<td>Missing</td>
		<td>Can cause problems with data type (turn a numerical column into a text column)</td>
		<td></td>
		<td>Avoid</td>
	</tr>
	<tr>
		<td>-,+</td>
		<td>Can cause problems with data type (turn a numerical column into a text column)</td>
		<td></td>
		<td>Never use</td>
    </tr>
</table>




7. Make it easy to combine your data with other datasets
--------------------------------------------------------

When working with large data collection efforts or when otherwise representing your data with limited space, it is common to develop codes or abbreviations which then appear in the data. In ecology and evolution codes often appear in place of site locations or taxonomy. Without clear definitions these codes are unintelligible and restrict reuse. As is common in standard database structure (cite?) we suggest the inclusion of linking tables which include fields for the item code and then additional fields which unambiguously describe the item. For species, this would include a table with the species codes followed by their most current family, genus, and species epithet. For site location, this would include a table with the site code followed by a GPS coordinate, spatial extent, temporal duration, and other appropriate site-specific details. These tables should be included directly with your data and mentioned as part of your meta-data. 


8. Perform basic quality control
--------------------------------

Data, just like any other scientific product, should undergo some level of quality control (Reichmann _et al._ 2011). At its most basic, quality control could consist of a few quick 'unit tests' of the data. These kind of tests can be easily implemented using spreadsheet, SQL, or coding software to scan the data for specific errors. For example, if a column should contain numeric values, check that there are no non-numeric values in the data. Ecological data often contains missing values. For good quality control, check that empty cells actually represent missing data, and not mistakes in data entry. If the values are truly missing, indicate using the appropriate null values. Scan your data for consistency in unit of measurement, data type (_e.g.,_ numeric, character), naming scheme (_e.g.,_ taxonomy, location). Problems or uncertainty in the data should be appropriately resolved before sharing your data, or described in detail in the metadata to avoid secondary use errors. 

A more rigorous level of quality control is to double-enter data and use SQL or coding software to check for mismatched lines between the two entries. Manual double-entry of data is more time-intensive than single entry, but is considered superior because it increases data accuracy by catching typographical errors, reader/recorder error, out-of-range values, and identifying questionable data (Paulsen 2012, Lampe _et al._ 1998). 

Before publishing the data, practice "data review". This can be as simple as asking a collaborator or another scientist unaffiliated with your specific project to scan your metadata and data. If they can't tell you what your data is about within 20 minutes, then you may need to recheck for the common problems and pitfalls listed above. You may have poor descriptions in your metadata, uninformative column names, or confusing data structure. Try to identify where your data is unclear or confusing and target those sections of your metadata or data structure. Informal data review may have added benefits of giving your data increased exposure and creating a culture of scientists who are better enabled to share and use other's data. 

__citations__
* Paulsen A., Overgaard S., Lauritsen J. M. 2012. Quality of Data Entry Using Single Entry, Double Entry and Automated Forms Processingâ€“An Example Based on a Study of Patient-Reported Outcomes. PLoS ONE. __7__:e35087. doi:10.1371/journal.pone.0035087

* Lampe, A. J. and J. M. Weiler. 1998. Data capture from the sponsors' and investigators' perspectives: balancing quality, speed, and cost. __32__:871-886.

9. Use an established repository
-------------------------------

[Kowalczyk & Shankar (2011)](http://onlinelibrary.wiley.com/doi/10.1002/aris.2011.1440450113/pdf) state that, in order for data sharing to be effective, data should be findable over time via consistent pointers (e.g. DOI's), accessible and protected from long-term loss, and of sufficient quality. 

* Long-term preservation, i.e. longevity:
	* Make sure your data has a digital object identifier (DOI). DOIs are permanent unique identifiers that are independent of physical location and ownership
	* Archives that provide DOI's: DataONE, figshare, Ecological Archives, Dryad
* Visibility
* Searchability
* Etc.
* Do we need to include a list of repos here? Yes. Plan to discuss general features of a few, and in relation to the above.

Ecological and evolutionary data take many forms. Molecular sequences, GIS and remotely sensed environmental data,
phylogenetic data, and census and capture data from populations and communities, among others are commonly used in
ecological and evolutionary research. Likewise, there are many online facilities, i.e. repositories, available for
sharing most any type of data. Some of these repositories, such as those hosting molecular sequence data (e.g. DDBJ,
GenBank, MG-RAST), are highly specific and standardized in terms of data type, format, etc. Other repositories host
a wide variety of ecological and evolutionary data and are much less standardized (e.g. Dryad, KNB, PANGAEA). 
Additionally, some repositories do not exclusively apply to biological sciences, but offer a host of powerful user
options such as version control and social networking (e.g. Figshare, Github). Repositories also vary in their terms
of use, rights maintained by providers, and in their accessibility to the public (Table ?). Choice of repository may
also depend on requirements of funding agencies and journals, as well as the preferences of data providers/owners for
specific rights and terms of use, i.e. licensing options.



<table>
<table border="1">
    <tr>
        <th>Repository</td>
		<th>License</td>
		<th>Assigns DOI</td>
        <th>Metadata</td>
        <th>Access</td>
		<th>Notes</td>
	</tr>
	<tr>
		<td>Dryad</td>
		<td>Creative Commons Zero (CC0)</td>
        <td>Yes</td>
		<td>suggested</td>
		<td>open</td>
		<td>Ecology & Evolution data</td>
	</tr>
	<tr>
		<td>Ecological Society of America (ESA) data registry</td>
		<td>N</td>
        <td>Yes</td>
		<td>compliant with EML</td>
		<td>varies among datasets</td>
		<td>data related to ESA publications</td>
	</tr>
    <tr>
		<td>Ecological Archives</td>
		<td>N</td>
        <td>Yes</td>
		<td>required</td>
		<td>open</td>
		<td>publishes supplemental material and data papers</td>
	</tr>
    <tr>
		<td>Knowledge Network for Biocomplexity (KNB)</td>
		<td>N</td>
        <td>Yes</td>
		<td>compliant with EML</td>
		<td>varies among datasets</td>
		<td>partners with ESA, NCEAS, DataONE</td>
	</tr>
	<tr>
		<td>National Center for Ecological Analysis and Synthesis (NCEAS)</td>
		<td>.</td>
        <td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
	</tr>
	<tr>
		<td>Paleobiology Database</td>
		<td>.</td>
        <td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
	</tr>
    <tr>
		<td>USA National Phenology Network</td>
		<td>.</td>
        <td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
	</tr>
	<tr>
		<td>Data Basin</td>
		<td>.</td>
        <td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
	</tr>
	<tr>
		<td>Pangaea</td>
		<td>.</td>
        <td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
	</tr>
	<tr>
		<td>CKAN Bioportal</td>
		<td>.</td>
        <td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
	</tr>
	<tr>
		<td>CKAN Climate Data</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
    </tr>
    <tr>
		<td>Global Biodiversity Information Facility (GBIF)</td>
		<td>.</td>
        <td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
    </tr>
    <tr>
		<td>Github</td>
		<td>.</td>
        <td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
    </tr>
    <tr>
		<td>Figshare</td>
		<td>.</td>
        <td>.</td>
		<td>.</td>
		<td>.</td>
		<td>.</td>
    </tr>
</table>



10. Use an established and liberal license 
-----------------------------------------

* http://blog.datadryad.org/2011/10/05/why-does-dryad-use-cc0/
* [About creative commons licenses] (http://creativecommons.org/licenses/)

Repositories have provisions for licensing (i.e. legally established intellectual property rights, copyright,
database right, terms of use) that are designed to protect those providing and hosting data and to clarify the
terms under which data is used (cite). Likewise, some repositories offer a variety of licensing options.
These can allow data sharers to share openly with the public, to allow only registered database users to access
their data, to make their data immediately available, to enforce an embargo time from publication, etc. Data sharers
should familiarize themselves with the policies and options of a particular repository.  



Concluding remarks
------------------

Data sharing has the potential to transform the way we conduct ecological and
evolutionary research. As a result there are an increasing number of initiatives
at the federal, funding agency, and journal levels to encourage or require the
sharing of the data associated with scientific research. However, making the
data available is only the first step. To make data sharing as useful as
possible it is necessary to make the data usable with as little effort as
possible. This allows scientists to spend their time doing science
rather than cleaning up data.

We have provided a list of 10 practices that require only a small additional
time investment but substantially improve the useability of data. Most of these
recommendations are simply good practice for working with data regardless of
whether that data is shared or not. This means that following these
recommendations (2-8) make the data more useful for those who collected it as
well as those who may use it in a secondary capacity. Well formatted and
structured data makes it easier to use the data in a variety of different
analysis programs. Well documented data makes it easier to use the data whether
the person using it is a lab mate, a meta-analyst, or the scientist that
collected the data a few years after it was collected. By following these
practices we can assure that the data collected in ecology and evolution can be
used to its full potential to improve our understanding of biological systems. 
